src/null-safety/migration-guide.md

```diff
@@ -1,23 +1,453 @@
+
+This section tells you how to
+check your package's dependencies
+using the `dart pub outdated` command in null-safety mode.
+
+{{ site.alert.warn }}
+  These instructions assume your code is under **source control**,
+  so that you can easily undo any changes.
+{{ site.alert.end }}
+
+### Switch to a 2.12 beta release
+
+**Switch to the latest beta release**
+of either the Dart SDK or the Flutter SDK.
+How you get the latest beta release depends on whether
+you use the Flutter SDK:
+
+* If you use the Flutter SDK,
+  switch to the beta channel:
+
+  ```terminal
+$ flutter channel beta
+$ flutter upgrade
+```
+* Otherwise, download a beta release
+  from the [Dart SDK archive][].
+
+[Dart SDK archive]: /tools/sdk/archive#beta-channel
+
+{{ site.alert.warn }}
+  If you return to working on production code, remember to
+  **switch back to a stable release**
+  (for example, by running `flutter channel stable`).
+{{ site.alert.end }}
+
+
+### Check dependency status
+
+Get the migration state of your package's dependencies,
+using the following command:
+
+```terminal
+$ dart pub outdated --mode=null-safety
+```
+
+If the output says **all the packages fully support null safety**,
+then you can start migrating.
+Otherwise, use the **Resolvable** column to find
+null-safe releases, if they exist.
+
+{{ site.alert.info }}
+  **Why do all dependencies need to support null safety?**
+  When all of an app's direct dependencies support null safety,
+  you can _run the app_ with sound null safety.
+  When all the dev dependencies support null safety,
+  you can _run tests_ with sound null safety.
+  You might also need null-safe dev dependencies for other reasons,
+  such as code generation.
+{{ site.alert.end }}
+
+Here's an example of the output for a simple package.
+The green checkmarked version for each package supports null safety:
+
+![Output of dart pub outdated, showing that examples_util has no releases with null safety support, but pedantic and test do](/null-safety/pub-outdated-output.png)
+
+Here are the steps to take before migrating this package to sound null safety:
+
+1. Wait for `examples_util` to be migrated.
+
+2. Update `pubspec.yaml` to use the latest compatible releases of
+   `pedantic`, `test`, and `examples_util`
+   (as listed in the **Resolvable** column).
+   For example, the new dev dependencies might look like this:
+   ```dart
+dev_dependencies:
+  examples_util: ^0.0.3-nullsafety.0
+  pedantic: ^1.10.0-nullsafety.3
+  test: ^1.16.0-nullsafety.9
+```
+
+3. Run `dart pub upgrade`.
+
+4. Run `dart pub outdated --mode=null-safety` again,
+   to make sure that all dependencies are null safe and up-to-date.
+
+
+## 2. Migrate {#step2-migrate}
+
+Most of the changes that your code needs to be null safe
+are easily predictable.
+For example, if a variable can be `null`,
+its type needs a `?` suffix.
+A named parameter that shouldn't be nullable
+needs to be marked `required`.
```

Comment:
Maybe link again to the "understanding null safety docs here"?

Reply:
+1